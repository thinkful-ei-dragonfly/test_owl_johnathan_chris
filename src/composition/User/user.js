import React from 'react';

function User(props) {
    let stageStatus; 
    if(props.onstage) {
        stageStatus = "on stage";
    } else if(!props.onstage && props.inSession) {
        stageStatus = "in session";
    } else {
        stageStatus = "left session";
    }
    return (
        <li className="user-participant">
            <div className="user-name">{props.name}</div>
            <img src={props.avatar}/>
            <span className="session-status">{stageStatus}</span>     
        </li>
        );
}

export default User;