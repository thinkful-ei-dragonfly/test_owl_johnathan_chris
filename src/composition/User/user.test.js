import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import User from './user';


  describe('User component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<User />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

    it('renders the UI as expected', () => {
    const tree = renderer
      .create(<User key={User.id} name="Johnathan" avatar='https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1' inSession={true} onStage={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
  });