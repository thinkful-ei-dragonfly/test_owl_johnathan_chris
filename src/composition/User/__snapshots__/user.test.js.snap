// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders the UI as expected 1`] = `
<li
  className="user-participant"
>
  <div
    className="user-name"
  >
    Johnathan
  </div>
  <img
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <span
    className="session-status"
  >
    in session
  </span>
</li>
`;
